### PR TITLE
Add `noaddon` option

### PR DESCRIPTION
**Pull request by sneridagh**
This is for the use case of the upcoming image. We need no addon in there to have a "clean" Volto if the image alone is run for demo purposes.

### CODE REVIEW

File: volto_addon/cookiecutter.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "email": "collective@plone.org",
   "github_organization": "collective",
   "npm_package_name": "{{ cookiecutter.addon_name }}",
+  "noaddon": false,
```
Comment (ericof) [inline volto_addon/cookiecutter.json:9]: Please, also add the user friendly version under `__prompts__`

File: volto_addon/hooks/post_gen_project.py [hunk 2]
```diff
@@ -38,8 +44,18 @@ def run_cmd(command: str, shell: bool, cwd: str) -> bool:
     return False if proc.returncode else True
 
 
+def remove(filepath):
+    if os.path.isfile(filepath):
```
Comment (ericof) [inline volto_addon/hooks/post_gen_project.py]: Use pathlib.Path

File: volto_addon/{{ cookiecutter.addon_name }}/package.json [hunk 3]
```diff
@@ -34,8 +34,12 @@
   },
   "dependencies": {
     "@plone/volto": "workspace:*",
+    {% if not cookiecutter.noaddon -%}
```
Comment (ericof) [inline volto_addon/{{ cookiecutter.addon_name }}/package.json]: Why not to use:
```
  "dependencies": {
    "@plone/volto": "workspace:*",
    "@plone/registry": "workspace:*"{%- if not cookiecutter.noaddon -%},
    "{{ cookiecutter.npm_package_name }}": "workspace:*"
    {%- endif %}
  },
```

File: volto_addon/{{ cookiecutter.addon_name }}/volto.config.js [hunk 4]
```diff
@@ -1,4 +1,8 @@
+{% if not cookiecutter.noaddon -%}
```
Comment (ericof) [inline volto_addon/{{ cookiecutter.addon_name }}/volto.config.js]: Why not to use:
```
const addons = [{%- if not cookiecutter.noaddon -%}'{{ cookiecutter.npm_package_name }}'{%- endif -%}];
```

**Review by ericof — CHANGES_REQUESTED** [review]
Left some comments about style, but I do have questions about having the whole noaddon option here. Let's talk a bit more

File: volto_addon/cookiecutter.json (hunk 1 above)
Comment (sneridagh) [reply volto_addon/cookiecutter.json:9]: Done

File: volto_addon/hooks/post_gen_project.py (hunk 2 above)
Comment (sneridagh) [reply volto_addon/hooks/post_gen_project.py]: Done

File: volto_addon/{{ cookiecutter.addon_name }}/package.json (hunk 3 above)
Comment (sneridagh) [reply volto_addon/{{ cookiecutter.addon_name }}/package.json]: Done

File: volto_addon/{{ cookiecutter.addon_name }}/volto.config.js (hunk 4 above)
Comment (sneridagh) [reply volto_addon/{{ cookiecutter.addon_name }}/volto.config.js]: Done

**Comment by sneridagh** [comment]
@ericof It's needed for the generation of the image without add-on. It can be even be a hidden feature. If not, when we generate the image, there will be always the default add-on in there.

**Comment by ericof** [comment]
Naming is hard :-)
I like the idea of a hidden option that would control this, let me try to
implement it

**Comment by sneridagh** [comment]
Let's do this in the image stage, not having a public facing way to do this.

**Pull request closed without merging**